### PR TITLE
Switch Apple benchmark workflow to use the generic ET benchmark iOS app

### PR DESCRIPTION
**Pull request by huydhn**
This requires @shoumikhin change in https://github.com/pytorch/executorch/pull/5208, so I will rebase and test it out again after https://github.com/pytorch/executorch/pull/5208 lands

### Testing

https://github.com/pytorch/executorch/actions/runs/10787058020

### CODE REVIEW

File: build/build_apple_llm_demo.sh [hunk 1]
```diff
@@ -7,50 +7,32 @@
...
-    -allowProvisioningUpdates \
-    DEVELOPMENT_TEAM=78E7V7QP35 \
-    CODE_SIGN_STYLE=Manual \
-    PROVISIONING_PROFILE_SPECIFIER=iLLaMA \
-    CODE_SIGN_IDENTITY="iPhone Distribution" \
-    CODE_SIGNING_REQUIRED=No \
-    CODE_SIGNING_ALLOWED=No
-fi
+ARTIFACTS_DIR_NAME="$1"
+APP_PATH="extension/apple/Benchmark/Benchmark"
```
Comment (guangy10) [inline build/build_apple_llm_demo.sh:11]: @shoumikhin what tokenizer is the app built with? w/ both bpe and tiktoken?

File: .github/workflows/apple-perf.yml [hunk 2]
```diff
@@ -245,8 +244,38 @@ jobs:
...
+        # app, we don't use this in CI and rely on AWS extra data parameter to make the model and the
+        # tokenizer available to the benchmark. This decouples the app and the model. We just need to
+        # create the directory here to pass the build
+        mkdir -p extension/apple/Benchmark/Models
         ${CONDA_RUN} --no-capture-output \
-          build/build_apple_llm_demo.sh ${{ matrix.tokenizer }} ${ARTIFACTS_DIR_NAME}
+          build/build_apple_llm_demo.sh ${ARTIFACTS_DIR_NAME}
 
   upload-ios-apps:
     needs: build-llm-demo
```
Comment (guangy10) [inline .github/workflows/apple-perf.yml:281]: Should we rename previous step to `build-benchmark-app`? Also should we update `ios-ipa-archive` and `ios-xctestrun-zip` to point to the new benchmark app

**Review by guangy10 — APPROVED** [review]
Left few comments, otherwise lgtm

**Review by shoumikhin — APPROVED** [review]

**Comment by huydhn** [comment]
The benchmark is running on device now, but it crashes once before being retrying successfully (weird) https://github.com/pytorch/executorch/actions/runs/10804559720/job/29970780023:

Here is the test output:

```
[DEVICEFARM] ########### Entering phase test ###########
 
[DeviceFarm] xcodebuild test-without-building -destination id=$DEVICEFARM_DEVICE_UDID -xctestrun $DEVICEFARM_TEST_PACKAGE_PATH/*.xctestrun -derivedDataPath $DEVICEFARM_LOG_DIR
Command line invocation:
    /Applications/Xcode_15.app/Contents/Developer/usr/bin/xcodebuild test-without-building -destination id=00008120-000424DC1122201E -xctestrun /tmp/devicefarm-workspace/execution-tvdc_j_e/test-package-ruiiiicu/Benchmark_Tests_iphoneos17.5-arm64.xctestrun -derivedDataPath /tmp/devicefarm-workspace/execution-tvdc_j_e/logs-3gvb1hak

User defaults from command line:
    IDEDerivedDataPathOverride = /tmp/devicefarm-workspace/execution-tvdc_j_e/logs-3gvb1hak
    IDEPackageSupportUseBuiltinSCM = YES

2024-09-10 21:07:33.547 xcodebuild[1200:8254]  DVTDevice: Error locating DeviceSupport directory using Optional("arm64e") or Optional("arm64e"): nilError
Test Suite 'All tests' started at 2024-09-10 21:07:34.514.
Test Suite 'Tests.xctest' started at 2024-09-10 21:07:34.514.
Test Suite 'Tests' started at 2024-09-10 21:07:34.514.
Test Case '-[Tests test_forward_aatp_data_llama2]' started.
2024-09-10 21:07:35.608 xcodebuild[1200:8342]  DVTDevice: Error locating DeviceSupport directory using Optional("arm64e") or Optional("arm64e"): nilError

Restarting after unexpected exit, crash, or test timeout in -[Tests test_forward_aatp_data_llama2]; summary will include totals from previous launches.

Test Suite 'Selected tests' started at 2024-09-10 21:07:36.076.
Test Suite 'Tests.xctest' started at 2024-09-10 21:07:36.077.
Test Suite 'Tests' started at 2024-09-10 21:07:36.077.
Test Case '-[Tests test_load_aatp_data_llama2]' started.
/Users/runner/work/executorch/executorch/pytorch/executorch/extension/apple/Benchmark/Tests/Tests.mm:58: Test Case '-[Tests test_load_aatp_data_llama2]' measured [Memory Peak Physical, kB] average: 122225.936, relative standard deviation: 0.000%, values: [122225.936000, 122225.936000, 122225.936000, 122225.936000, 122225.936000], performanceMetricID:com.apple.dt.XCTMetric_Memory.physical_peak, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.000, maxStandardDeviation: 0.000
/Users/runner/work/executorch/executorch/pytorch/executorch/extension/apple/Benchmark/Tests/Tests.mm:58: Test Case '-[Tests test_load_aatp_data_llama2]' measured [Memory Physical, kB] average: 0.000, relative standard deviation: 0.000%, values: [0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.dt.XCTMetric_Memory.physical, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.000, maxStandardDeviation: 0.000
/Users/runner/work/executorch/executorch/pytorch/executorch/extension/apple/Benchmark/Tests/Tests.mm:58: Test Case '-[Tests test_load_aatp_data_llama2]' measured [Clock Monotonic Time, s] average: 0.000, relative standard deviation: 26.339%, values: [0.000001, 0.000001, 0.000001, 0.000001, 0.000001], performanceMetricID:com.apple.dt.XCTMetric_Clock.time.monotonic, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.000, maxStandardDeviation: 0.000
Test Case '-[Tests test_load_aatp_data_llama2]' passed (0.152 seconds).
Test Suite 'Tests' failed at 2024-09-10 21:07:36.229.
	 Executed 2 tests, with 1 failure (0 unexpected) in 0.152 (0.152) seconds
Test Suite 'Tests.xctest' failed at 2024-09-10 21:07:36.229.
	 Executed 2 tests, with 1 failure (0 unexpected) in 0.152 (0.152) seconds
Test Suite 'Selected tests' failed at 2024-09-10 21:07:36.229.
	 Executed 2 tests, with 1 failure (0 unexpected) in 0.152 (0.153) seconds
2024-09-10 21:07:42.661 xcodebuild[1200:8040] [MT] IDETestOperationsObserverDebug: 17.357 elapsed -- Testing started completed.
2024-09-10 21:07:42.661 xcodebuild[1200:8040] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2024-09-10 21:07:42.661 xcodebuild[1200:8040] [MT] IDETestOperationsObserverDebug: 17.357 sec, +17.357 sec -- end

Test session results, code coverage, and logs:
	/tmp/devicefarm-workspace/execution-tvdc_j_e/logs-3gvb1hak/Logs/Test/Test-Benchmark-2024.09.10_21-07-25--0700.xcresult

Failing tests:
	-[Tests test_forward_aatp_data_llama2]

** TEST EXECUTE FAILED **
```

@shoumikhin has some leads on the cause of the crash:

```
et_view.cpp:88] In function et_view(), assert failed: self.numel() == out.numel()
```

So, I guess we need to figure this out and fix the issue before this can be landed.

**Comment by guangy10** [comment]
Is this PR ready to test other models w/ different delegates. If the crash is llama specific, we can verify the app with nongenai models in parallel.

**Comment by huydhn** [comment]
D62511672 works https://github.com/pytorch/executorch/actions/runs/10818420721/job/30015121518 :)

**Pull request closed without merging**